### PR TITLE
TimeSeries: Fix rendering when zooming to time ranges between datapoints

### PR DESCRIPTION
**Pull request by leeoniya**
this bumps uPlot to 1.6.23 and removes some additional work-arounds we had on our side.

### CODE REVIEW

File: packages/grafana-ui/src/components/uPlot/config/UPlotSeriesBuilder.ts [hunk 1]
```diff
@@ -254,8 +254,8 @@ function mapDrawStyleToPathBuilder(
     builders = {
       linear: pathBuilders.linear!(),
       smooth: pathBuilders.spline!(),
-      stepBefore: pathBuilders.stepped!({ align: -1 }),
-      stepAfter: pathBuilders.stepped!({ align: 1 }),
+      stepBefore: pathBuilders.stepped!({ align: -1, extend: true }),
+      stepAfter: pathBuilders.stepped!({ align: 1, extend: true }),
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/uPlot/config/UPlotSeriesBuilder.ts:258]: these bits will cause stepped interpolation to extend beyond the data to the edges of the plotting area:

image

**Comment by leeoniya** [comment]
@ryantxu think this addresses https://github.com/grafana/grafana/pull/57798#issuecomment-1297734092

http://localhost:3000/d/pqnrfd4Vz/panel-tests-timeseries-out-of-range?orgId=1&from=1662008703623&to=1662014644897

image

**Comment by toddtreece** [comment]
@leeoniya tested locally, and it works for me. thanks!

**Comment by leeoniya** [comment]
not a breaking change (internal use, and pretty fresh API that's not needed anymore due to fix landed in dependency)

image

**Review by ryantxu — APPROVED** [review]
You'll kill me for the inconsistent message.... but...

since this fixes a bug (zooming past point range), can we backport to 9.3?  We would need to add the label "add-to-changelog" if we did

**Review by nmarrs — APPROVED** [review]
🥇

**Pull request merged**